src/service.ts

```diff
@@ -207,36 +209,46 @@ export class DebugService implements IDebugger {
     return reply.body.variables;
   };
 
-  getBreakpoints = (): DebugProtocol.SourceBreakpoint[] => {
-    return this._model.breakpointsModel.breakpoints.map(breakpoint => {
-      return {
-        line: breakpoint.line
-      };
-    });
-  };
-
   setBreakpoints = async (
     breakpoints: DebugProtocol.SourceBreakpoint[],
     path: string
   ) => {
     // Workaround: this should not be called before the session has started
     const client = this.session.client as IClientSession;
     await client.ready;
-    await this.session.sendRequest('setBreakpoints', {
+    return await this.session.sendRequest('setBreakpoints', {
       breakpoints: breakpoints,
       source: { path },
       sourceModified: false
     });
   };
 
-  updateBreakpoints = async () => {
+  updateBreakpoints = async (breakpoints: Breakpoints.IBreakpoint[]) => {
     if (!this.session.isStarted) {
       return;
     }
     const code = this._model.codeValue.text;
     const dumpedCell = await this.dumpCell(code);
-    const breakpoints = this.getBreakpoints();
-    await this.setBreakpoints(breakpoints, dumpedCell.sourcePath);
+    const sourceBreakpoints = Private.toSourceBreakpoints(breakpoints);
+    const reply = await this.setBreakpoints(
+      sourceBreakpoints,
+      dumpedCell.sourcePath
+    );
+    let kernelBreakpoints = reply.body.breakpoints.map(breakpoint => {
+      return {
+        ...breakpoint,
+        active: true,
+        source: { path: this.session.client.name }
+      };
+    });
+
+    // filter breakpoints with the same line number
```

Comment:
In theory an editor can display multiple breakpoints on the same line. But we can skip that case for now.

Reply:
What use case do you have in mind for setting several breakpoints on the same line?

Reply:
For example in the dev tools:

![chrome-dev-tools-inline-breakpoints](https://user-images.githubusercontent.com/591645/67937949-d1034d00-fbce-11e9-9936-bdfd80d9ec56.gif)

But yeah as mentioned above it's fine to not do that and skip this feature for this debugger.